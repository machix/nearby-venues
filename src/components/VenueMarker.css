.venue-marker {
  padding: 5px;
  position: relative;
  color: white;
  box-shadow: 0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24);
  width: 70px;
  left: -50%;
  transform: translateY(-100%);
  display: flex;
  align-items: center;
  justify-content: space-between;
  z-index: 1;
}

.venue-rating {
  border: 0.5px solid white;
  border-radius: 2px;
  padding: 2px 5px;
}

.venue-tier {
}

.bottom-arrow {
  position: absolute;
  left: calc(50% - 5px);
  top: 100%;
  border-left: 5px solid transparent;
  border-right: 5px solid transparent;
  border-top: 5px solid #000;
  clear: both;
}
.venue-marker:hover {
  z-index: 99;
  background-color: orange!important;
}
.venue-marker:hover .bottom-arrow {
  border-top-color: orange!important;
}

.marker-details {
  visibility: hidden;
  position: absolute;
  background-color: white;
  color: black;
  opacity: 0;
  bottom: 100%;
  left: calc(50% - 125px);
  box-sizing: border-box;
  box-shadow: 0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24);
  width: 250px;
  overflow: hidden;
  padding: 0;
  transition: opacity 0.5s, padding 0.2s;
}

.venue-marker:hover .marker-details{
  display: block;
  padding: 10px;
  opacity: 1;
  visibility: visible;
}

.venue-name {
  font-weight: bold;
  margin: 0;
  padding-bottom: 5px;
  border-bottom: 1px solid #e2e2e2;
  margin-bottom: 5px;
  display: block;
  color: black;
}
.radius-select {
  position: absolute;
  right: 10px;
  top: 10px;  
}
.marker-detail-content {
  display: flex;
  justify-content: space-between;
  align-items: center;
}
.marker-detail-content .left-content {
  display: flex;
  flex-direction: column;
  align-items: space-between;
}
.venue-tag {
  padding: 5px 10px;
  margin: 0;
  background-color: orange;
  border-radius: 5px;
  color: white;
  display: inline-block;
  background: #adadad;
  font-weight: 100;
}

.address-text {
  margin: 5px;
}