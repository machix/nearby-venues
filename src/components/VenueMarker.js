import React from 'react';
import './VenueMarker.css';

const VenueMarker = ({ rating = '-', tier, ratingColor, name, tag, formattedAddress, lat, lng, phone, url }) => (
  <div style={{ backgroundColor: '#' + ratingColor }} className='venue-marker'>
    <span className='venue-rating'>{rating}</span>
    <span className='venue-tier'>{new Array(tier).fill('$').join(' ')}</span>
    <span className='bottom-arrow' style={{ borderTopColor: '#' + ratingColor }}></span>
    <div className='marker-details'>
      <a href={`https://www.google.com/maps/?q=${lat},${lng}`} target='_blank' className='venue-name'>{name}</a>
      <p className='venue-tag'>{tag}</p>
      <div className='marker-detail-content'>
        <div className='left-content'>
          <p className='venue-rating-detail'>Rating: {rating}</p>
          <a target='_blank' href={`http://maps.google.com/?daddr=${lat},${lng}`} className="btn" type="button"><span>Navigate</span></a>
          <a target='_blank' href={`tel:${phone}`} className="btn" type="button"><span>Call</span></a>
          <a target='_blank' href={url} className="btn" type="button"><span>Website</span></a>
        </div>
        <div className='address-container'>
          <p>Address: </p>
          {formattedAddress.map((address, i) => <p key={i} className='address-text'>{address}</p>)}
        </div>
      </div>
    </div>
  </div>
);

export default VenueMarker;