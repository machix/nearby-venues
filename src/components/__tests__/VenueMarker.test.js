import React from 'react';
import VenueMarker from '../VenueMarker';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const props = {
    rating: '-',
    tier: 4,
    ratingColor: 'green',
    name: 'john doe',
    tag: 'hotel',
    formattedAddress: ['hi', 'abc'],
    lat: 13.7563,
    lng: 100.5018,
    phone: '9812472772',
    url: 'google.com'
  }
  const tree = renderer.create(<VenueMarker {...props} />);
  expect(tree).toMatchSnapshot();
});
