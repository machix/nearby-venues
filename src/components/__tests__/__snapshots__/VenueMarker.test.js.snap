// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="venue-marker"
  style={
    Object {
      "backgroundColor": "#green",
    }
  }
>
  <span
    className="venue-rating"
  >
    -
  </span>
  <span
    className="venue-tier"
  >
    $ $ $ $
  </span>
  <span
    className="bottom-arrow"
    style={
      Object {
        "borderTopColor": "#green",
      }
    }
  />
  <div
    className="marker-details"
  >
    <a
      className="venue-name"
      href="https://www.google.com/maps/?q=13.7563,100.5018"
      target="_blank"
    >
      john doe
    </a>
    <p
      className="venue-tag"
    >
      hotel
    </p>
    <div
      className="marker-detail-content"
    >
      <div
        className="left-content"
      >
        <p
          className="venue-rating-detail"
        >
          Rating: 
          -
        </p>
        <a
          className="btn"
          href="http://maps.google.com/?daddr=13.7563,100.5018"
          target="_blank"
          type="button"
        >
          <span>
            Navigate
          </span>
        </a>
        <a
          className="btn"
          href="tel:9812472772"
          target="_blank"
          type="button"
        >
          <span>
            Call
          </span>
        </a>
        <a
          className="btn"
          href="google.com"
          target="_blank"
          type="button"
        >
          <span>
            Website
          </span>
        </a>
      </div>
      <div
        className="address-container"
      >
        <p>
          Address: 
        </p>
        <p
          className="address-text"
        >
          hi
        </p>
        <p
          className="address-text"
        >
          abc
        </p>
      </div>
    </div>
  </div>
</div>
`;
